scripts/zones/Arrapago_Reef/mobs/Medusa.lua

```diff
@@ -1,28 +1,45 @@
 -----------------------------------
 -- Area: Arrapago Reef
 -- NPC:  Medusa
+-- TODO: resists, attack/def boosts
 -----------------------------------
 
 require("scripts/globals/titles");
+require("scripts/zones/Arrapago_Reef/MobIDs");
+require("scripts/zones/Arrapago_Reef/TextIDs");
 
 -----------------------------------
 -- onMobSpawn Action
 -----------------------------------
 
 function onMobSpawn(mob)
+    mob:setLocalVar("eeshpp", math.random(5,99)); -- Uses EES randomly during the fight
 end;
 
 -----------------------------------
 -- onMobEngaged Action
 -----------------------------------
 
-function onMobEngaged(mob,target)
+function onMobEngaged(mob, target)
+    target:showText(mob, MEDUSA_ENGAGE);
+    SpawnMob(ExonOne,   180):updateEnmity(target);
+    SpawnMob(ExonTwo,   180):updateEnmity(target);
+    SpawnMob(ExonThree, 180):updateEnmity(target);
+    SpawnMob(ExonFour,  180):updateEnmity(target);
+end;
 
-	SpawnMob(16998863,180):updateEnmity(target);
-	SpawnMob(16998864,180):updateEnmity(target);
-	SpawnMob(16998865,180):updateEnmity(target);
-	SpawnMob(16998866,180):updateEnmity(target);
+-----------------------------------
+-- onMobFight Action
+-----------------------------------
 
+function onMobFight(mob, target)
+    local HPP = mob:getHPP();
+    if (mob:getLocalVar("usedees") == 0) then
+        if (HPP <= mob:getLocalVar("eeshpp")) then
+            mob:useMobAbility(479); -- Eagle Eye Shot

```

Comment:
Well I disagree using an id is not better. This will force every scripter to look up the correct mobskill and has the potential to use the wrong one, in fact in this case `479` is the wrong mobskill.

All this two hour stuff won't get solved in this PR, so let's figure it out later.

@nesstea I'm about 100% certain `479` is not the correct mob skill. Run `@injectaction 11 19` with Medusa selected, it will most likely run the wrong animation.